### PR TITLE
Feat/examples create inner asset

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+ Add coordinator.py. This application will handle new appl, asset and Log id of them. Have 2 mode use group txn and use inner txn(Check code for more details)
+ Add python `pyteal-utils` package. 
+ deploy script and run scripts.   

## Blocking
- When run executeTx with group transaction, method doesn't return array which is receipts of group txn. 

**Please review logic of smart contract first.**

### CODE REVIEW

File: examples/create-inner-asset/README.md [hunk 1]
```diff
@@ -1 +1,3 @@
 # Create-inner-asset
+
```
Comment (robert-zaremba) [inline examples/create-inner-asset/README.md:2]: let's rename the directory and example to "inner-tx_create_assets"

File: examples/create-inner-asset/assets/worker_app.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from pyteal import *
```
Comment (robert-zaremba) [inline examples/create-inner-asset/assets/worker_app.py:1]: 1. please add a doc comment to what this app is doing, also say that his is demo app and doesn't have security measures. Make similar description in the README file.
2. maybe let's rename to something more meaningful? How about "logger.py"?
3. Also, I think we can simplify and remove not used code (eg delete check ...)

File: examples/create-inner-asset/assets/master_app.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+from pyteal import *
+
```
Comment (robert-zaremba) [inline examples/create-inner-asset/assets/master_app.py]: 1. please add a doc comment to what this app is doing, also say that his is demo app and doesn't have security measures. Make similar description in the README file.
2. explain the composition of transactions, this app should be called as a 3rd transaction, right?
3. maybe let's rename to something more meaningful? How about "coordinator.py"?

File: examples/create-inner-asset/assets/master_app.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+@Subroutine(TealType.none)
+def call_logs():
+    app_create, asset_create, proxy_action = Gtxn[0], Gtxn[1], Gtxn[2]
```
Comment (robert-zaremba) [inline examples/create-inner-asset/assets/master_app.py:6]:
```suggestion
    app_create, asset_create, this_tx = Gtxn[0], Gtxn[1], Gtxn[2]
```

File: examples/create-inner-asset/assets/master_app.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    validate_txns = And(
```
Comment (robert-zaremba) [inline examples/create-inner-asset/assets/master_app.py:7]:
```suggestion
    validate_txns = And(
        Txn.group_index() == 2,  // check that this tx is the third transaction
```

File: examples/create-inner-asset/assets/worker_app.py (hunk 2 above)
Comment (vuvth) [reply examples/create-inner-asset/assets/worker_app.py]: Done.

File: examples/inner-tx-create-assets/scripts/run/useInnerTxn.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		},
+	};
+
+	const txReceipt = await executeTx(deployer, [masterTxnParam]);
+
+	// only one transaction
+	const logs = txReceipt[0].logs;
+	// Log asset id
+	console.log("New asset Id =", decodeValue(logs[0]));
+	// Log application id
```
Comment (sczembor) [inline examples/inner-tx-create-assets/scripts/run/useInnerTxn.js:29]: Do we need those comments? I think is self explanatory

File: examples/inner-tx-create-assets/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Inner-tx-create-assets 
+
+**Note: This is demo application, please don't use it in production. Thanks**
```
Comment (sczembor) [inline examples/inner-tx-create-assets/README.md:3]:
```suggestion
**Note: This is a demo application, please don't use it in production. Thanks**
```

File: examples/inner-tx-create-assets/assets/coordinator.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
+"""
+NOTE: This is demo contracts, doesn't have security measures. Please don't use it in product.
```
Comment (sczembor) [inline examples/inner-tx-create-assets/assets/coordinator.py:2]:
```suggestion
NOTE: This is  a demo contract, doesn't have security measures. Please don't use it in production
```

File: examples/inner-tx-create-assets/assets/coordinator.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+This smart contract doing:
+
+1. Demo create application, asset and use new appl and asset id after we create them
```
Comment (sczembor) [inline examples/inner-tx-create-assets/assets/coordinator.py:6]:
```suggestion
1. Creates a demo application, an asset and uses new appl and asset id after we created them 
```
This is a bit confusing I am not sure what you meant here. Could you check it one more time?

File: examples/inner-tx-create-assets/assets/coordinator.py [hunk 10]
```diff
@@ -0,0 +1,131 @@
+"""
+NOTE: This is demo contracts, doesn't have security measures. Please don't use it in product.
+
+This smart contract doing:
+
+1. Demo create application, asset and use new appl and asset id after we create them 
+   by using group transaction.
+
+2. Demo create application, asset and use new appl and asset id after we create them
```
Comment (sczembor) [inline examples/inner-tx-create-assets/assets/coordinator.py:9]: same as above

File: examples/inner-tx-create-assets/scripts/run/useGroupTxn.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+const { bytesToBigInt } = require("algosdk");
+const { accounts, decodeValue } = require("../utils");
+
+// Deploy new application
+async function run(runtimeEnv, deployer) {
+	const { creator } = accounts(deployer);
+
+	const proxyAppInfo = deployer.getApp("coordinator");
+
+	// first tx in group: deploy new app app
```
Comment (sczembor) [inline examples/inner-tx-create-assets/scripts/run/useGroupTxn.js:12]:
```suggestion
	// first tx in group: deploy new app
```

**Review by sczembor — APPROVED** [review]
Left few comments

File: examples/inner-tx-create-assets/assets/coordinator.py (hunk 9 above)
Comment (vuvth) [reply examples/inner-tx-create-assets/assets/coordinator.py:6]: Changed, please check @sczembor

File: examples/inner-tx-create-assets/scripts/run/useInnerTxn.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+		payFlags: {
+			totalFee: 3000,
+		},
+	};
+
+	const txReceipt = await executeTx(deployer, [masterTxnParam]);
+
+	// only one transaction
+	const logs = txReceipt[0].logs;
+	// Log asset id
```
Comment (vuvth) [inline examples/inner-tx-create-assets/scripts/run/useInnerTxn.js:27]:
```suggestion
```

File: examples/inner-tx-create-assets/scripts/run/useInnerTxn.js (hunk 6 above)
Comment (vuvth) [inline examples/inner-tx-create-assets/scripts/run/useInnerTxn.js:29]:
```suggestion
```

**Pull request merged**